src/main/java/com/endava/soa_5/actions/EditUserDataPageActions.java

```diff
@@ -0,0 +1,81 @@
+package com.endava.soa_5.actions;
+
+import com.endava.soa_5.abstract_classes.BaseClass;
+import com.endava.soa_5.page_objects.LogInPageObject;
+import com.endava.soa_5.page_objects.MyAccountPage.CustomerInfoPage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class EditUserDataPageActions extends BaseClass {
+    Logger log = LoggerFactory.getLogger(this.getClass());
+    private CustomerInfoPage customerInfoPage;
+    private LogInPageObject logInPageObject;
+
+    public EditUserDataPageActions() {
+        customerInfoPage = new CustomerInfoPage();
+        logInPageObject = new LogInPageObject(driver);
+    }
+
+    public void clickAccountNameButton() {
+        logInPageObject.getAccountName().click();
+    }
+
+    public void fillInCustomerInfo(Map<String, String> userData) {
+        if (userData.get("Gender") != null) {
+            if (userData.get("Gender").equalsIgnoreCase("M")) {
+                customerInfoPage.getMaleRadioButton().click();
+            } else if (userData.get("Gender").equalsIgnoreCase("F")) {
+                customerInfoPage.getFemaleRadioButton().click();
+            }
+        }
+        if (userData.get("First name") != null) {
+            customerInfoPage.getFirstNameTextBox().clear();
+            customerInfoPage.getFirstNameTextBox().sendKeys(userData.get("First name"));
+        }
+        if (userData.get("Last name") != null) {
+            customerInfoPage.getLastNameTextBox().clear();
+            customerInfoPage.getLastNameTextBox().sendKeys(userData.get("Last name"));
+        }
+    }
+
+    public String getGenderCheck(String gender) {
+        if (gender.equalsIgnoreCase("M")) {
+            return customerInfoPage.getMaleRadioButton().getAttribute("checked");
+        } else if (gender.equalsIgnoreCase("F")) {
+            return customerInfoPage.getFemaleRadioButton().getAttribute("checked");
+        }
+        return null;
```

Comment:
I will throw here an exception because of an unknown argument that you can't handle.